File: Structural - Exercises/Solutions-Leonie/Bridge/protocols/imap.py
from protocols.protocol_interface import ProtocolInterface

class IMAPProtocol(ProtocolInterface):
    def initialize(self):
        print("IMAP: Initializing email protocol")

    def send_email(self, message: str):
        print(f"IMAP: Sending email: {message}")
        self.provider.process_email(message)

    def shutdown(self):
        print("IMAP: Shutting down protocol")

File: Lesson.03 - Design Patterns - Structural/Bridge/IMAP.java
public class IMAP implements EmailProtocol {
    @Override
    public void initialize() {
        System.out.println("IMAP: Initializing message access protocol");
    }

    @Override
    public void sendEmail(String message) {
        System.out.println("IMAP: Sending email: " + message);
    }

    @Override
    public void shutdown() {
        System.out.println("IMAP: Shutting down protocol");
    }
}
